peekingduck/pipeline/nodes/model/huggingface_hubv1/huggingface_hub_model.py

```diff
@@ -276,8 +312,111 @@ def _postprocess(
+        """Creates the specified Hugging Face model from pretrained weights.
+        In addition, also sets up cache directory paths, maps `detect` list to
+        numeric detect IDs, and attempts to setup inference on GPU.
+        """
+        model_dir = self.prepare_cache_dir(model_type.split("/"))
+
+        detector_config = AutoConfig.from_pretrained(model_type, cache_dir=model_dir)
+        detect_ids = self._map_detect_ids(detector_config, self.config["detect"])
+
+        try:
+            self.adaptor = PanopticSegmenter(model_type, model_dir, self.mask_threshold)
+        except ValueError:
+            self.adaptor = InstanceSegmenter(model_type, model_dir, self.mask_threshold)
+        self.device = self._init_device()
+        self.detect_ids = detect_ids
+
+    def _log_model_configs(self) -> None:
+        """Prints the loaded model's settings."""
+        self.logger.info(
+            "Hugging Face model loaded with the following configs:\n\t"
+            f"Model type: {self.config['model_type']}\n\t"
+            f"IDs being detected: {self.detect_ids}\n\t"
+            f"Mask threshold: {self.mask_threshold}\n\t"
+            f"Score threshold: {self.score_threshold}"
+        )
+
+    def _postprocess(
+        self, result: Dict[str, torch.Tensor], image_shape: Tuple[int, int]
+    ) -> Tuple[np.ndarray, ...]:
+        """Post processes detection result. Filters detection results by
+        confidence score and discards detections which do not contain objects
+        of interest based on ``self._detect_ids``.
+
+        Args:
+            result (Dict[str, torch.Tensor]): A dictionary containing the model
+                output, with the keys: "boxes", "labels", "masks, and "scores".
+            image_shape (Tuple[int, int]): The height and width of the input
+                image.
+
+        Returns:
+            (Tuple[np.ndarray, np.ndarray, np.ndarray]): Returned tuple
+            contains:
+            - An array of detection bboxes
+            - An array of human-friendly detection class names
+            - An array of detection scores
+            - An array of binary instance masks
+        """
+        if result["masks"].size(0) == 0:
+            return (
+                np.empty((0, 4), dtype=np.float32),
+                np.empty(0),
+                np.empty(0, dtype=np.float32),
+                np.empty((0, 0, 0), dtype=np.uint8),
+            )
+        detect_filter = torch.isin(result["labels"], self._detect_ids)
+        score_filter = result["scores"] > self.score_threshold
+        for key in result:
+            result[key] = (
+                result[key][detect_filter & score_filter].cpu().detach().numpy()
+            )
+        bboxes = xyxy2xyxyn(result["boxes"], *image_shape)
+        classes = np.array([self.adaptor.id2label[int(i)] for i in result["labels"]])
+        return bboxes, classes, result["scores"], result["masks"].squeeze(1)
```

Comment:
Not really a one-to-one pairing, `InstanceSegmentationModel` can use either a `PanopticSegmenter` or a `InstanceSegmenter` adaptor.

Since it's possible for PeekingDuck to have panoptic segmentation models nodes in the future, these models and adaptors are kept in separater files.